File: sub.py
import findspark
findspark.init()
from pyspark.sql.session import SparkSession
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
import pyspark
import time
sc = pyspark.SparkContext()

spark = SparkSession(sc)

schema = StructType() \
  .add("author", StringType()) \
  .add("permlink", StringType()) \
  .add("last_update", StringType()) \
  .add("keyword", StringType()) \
  .add("id", StringType()) \
  .add("category", StringType()) \
  .add("title", StringType()) \
  .add("body", StringType()) \
  .add("created", StringType()) \
  .add("net_votes", IntegerType())

df = spark \
  .readStream \
  .format("kafka") \
  .option("kafka.bootstrap.servers", "localhost:9092") \
  .option("subscribe", "steemit") \
  .load() \
  .select(col("key").cast("string"), from_json(col("value").cast("string"), schema).alias("parsed_value"))

df_str = df.select(col("key"), col("parsed_value.*"))
output = df_str.writeStream.format("console").option("truncate", "false").trigger(processingTime='1 seconds').start()

i = 0
while True:
    time.sleep(1)
    i += 1

File: sub.scala
import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.streaming.ProcessingTime
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import java.io.FileInputStream
import java.util.Properties

object sub {
  def main(args: Array[String]): Unit = {
    val p = new Properties()
    p.load(new FileInputStream("gazua.ini"))
    val url = p.getProperty("url")
    val user = p.getProperty("user")
    val pwd  = p.getProperty("pwd")
    
    val writer = new JDBCSink(url, user, pwd)
    val spark = SparkSession
      .builder()
      .appName("Steemit Subscriber")
      .master("local[4]")
      .getOrCreate()

    import spark.implicits._

    val schema = StructType(
      StructField("author", StringType, true) ::
      StructField("permlink", StringType, true) ::
      StructField("last_update", StringType, true) ::
      StructField("keyword", StringType, true) ::
      StructField("id", StringType, true) ::
      StructField("category", StringType, true) ::
      StructField("title", StringType, true) :: 
      StructField("body", StringType, true) :: 
      StructField("created", StringType, true) :: 
      StructField("net_votes", IntegerType, true) :: Nil
    )

    val df = spark
      .readStream
      .format("kafka")
      .option("kafka.bootstrap.servers", "localhost:9092")
      .option("subscribe", "steemit")
      .load()

    val values = df.select($"key" cast "string", $"value" cast "string").select($"key", from_json($"value", schema) as "data").select("key", "data.*")

    values.writeStream
      .foreach(writer)
      .outputMode("append")
      .trigger(ProcessingTime("10 seconds"))
      .start()
      .awaitTermination()
  }
}
